fdbserver/SimulatedCluster.actor.cpp

```diff
@@ -2064,16 +2072,29 @@ void setupSimulatedSystem(std::vector<Future<Void>>* systemActors,
 
 	simconfig.db.tenantMode = tenantMode;
 	simconfig.db.encryptionAtRestMode = EncryptionAtRestMode::DISABLED;
-	if (!testConfig.encryptModes.empty()) {
-		simconfig.db.encryptionAtRestMode =
-		    EncryptionAtRestMode::fromString(deterministicRandom()->randomChoice(testConfig.encryptModes));
-	} else if (!testConfig.disableEncryption && deterministicRandom()->coinflip()) {
-		if (tenantMode == TenantMode::DISABLED || tenantMode == TenantMode::OPTIONAL_TENANT ||
-		    deterministicRandom()->coinflip()) {
-			// optional and disabled tenant modes currently only support cluster aware encryption
-			simconfig.db.encryptionAtRestMode = EncryptionAtRestMode::CLUSTER_AWARE;
+	// TODO: Remove check on the ENABLE_ENCRYPTION knob once the EKP can start using the db config
+	if (!testConfig.disableEncryption && (SERVER_KNOBS->ENABLE_ENCRYPTION || !testConfig.encryptModes.empty())) {
```

Comment:
looks like we can remove `disableEncryption` and replace it by `encryptModes = ["disabled"]` in test config.

Reply:
good point will change

Reply:
Actually i took a look we use this to disable the knobs currently. I'll create a todo to remove this in a final pr where we rip out the knobs